code/modules/mob/mob.dm

```diff
@@ -567,9 +583,21 @@
 	to_chat(src, examine_block(span_infoplain(result_combined)))
 	SEND_SIGNAL(src, COMSIG_MOB_EXAMINATE, examinify)
 
-/mob/proc/blind_examine_check(atom/examined_thing)
+//This is a proc for worn item examines. See /obj/item/proc/get_examine_line
+/mob/proc/run_worn_examinify_checks(atom/examinify)
+	// Allow examine from up to 4 tiles away
+	if(!isobserver(usr) && !(usr in viewers(4, get_turf(examinify))))
+		to_chat(usr, span_warning("You are too far away!"))
+		return FALSE
+	if(is_blind())
+		//blind_examine_check has some funky item movement stuff going on, so we just block blind examines
+		to_chat(usr, span_warning("You can't feel those details!"))
+		return FALSE
 	return TRUE
```

Comment:
Check for the user's ability to read, if they can't read something then they wouldn't be able to inspect something closer. You may have to update the proc to be generic since `can_read` sends its own to_chat messages. I would add a silent flag to it and then call silent true, and then have a 'you can't make anything out' if it fails.

Reply:
1d46f3ee13f5282f10061c4d83a4d28ba4bc1c0c